pkg/apis/core/validation/validation.go

```diff
@@ -5364,9 +5368,12 @@ func ValidateResourceRequirements(requirements *core.ResourceRequirements, fldPa
 		} else if !helper.IsOvercommitAllowed(resourceName) {
 			allErrs = append(allErrs, field.Required(limPath, "Limit must be set for non overcommitable resources"))
 		}
-		if helper.IsHugePageResourceName(resourceName) {
-			reqContainsHugePages = true
+
+		reqContainsHugePages, err = validateResourceQuantityHugePageValue(resourceName, quantity, opts)
```

Comment:
This overwrites an existing `reqContainsHugePages` `true` value with `false` if the last resource validated is not a hugepages resource. This would be more straightforward like this:

```go
if helper.IsHugePageResourceName(resourceName) {
  reqContainsHugePages = true
  if err := validateResourceQuantityHugePageValue(resourceName, quantity, opts); err != nil {
    allErrs = append(allErrs, field.Invalid(fldPath, quantity.String(), err.Error()))
  }
}
```

Please add a unit test that would have caught this: pod req that has hugepages, then several non-hugepages/non-cpu/non-memory reqs (several are required because iteration order is non-deterministic)

Reply:
updated. see unit test details TestValidateResourceRequirements @liggitt

Reply:
same here, did you push the change up?

Reply:
sorry, I fogot to updated the code. Please check again.